### PR TITLE
Added OrbitZoomCamera

### PR DESCRIPTION
**Pull request by stjahns**
Provides a 3dsMax/Blender style camera that orbits around a target
position, zoom in or out from the target position, or pan relative
to the camera's orientation

### CODE REVIEW

File: src/camera.rs [hunk 1]
```diff
@@ -11,6 +11,8 @@ use vecmath::{
 };
 use vecmath::col_mat4_mul as mul;
 use vecmath::consts::Radians;
+use cgmath::{ Quaternion, Rotation, FixedArray, BaseFloat };
+use cgmath::Vector3 as CgVector3;
```
Comment (stjahns) [inline src/camera.rs]: Is there a good reason to not switch completely to cgmath in this lib? Quaternions are kinda nice for a camera library.

File: src/orbit_zoom_camera.rs [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    keys: Keys,
+}
+
+
+impl OrbitZoomCamera {
+
+    ///
+    /// Create a new OrbitZoomCamera targeting the given coordinates
+    /// 
+    pub fn new(target: [f32; 3], settings: OrbitZoomCameraSettings) -> OrbitZoomCamera {
```
Comment (stjahns) [inline src/orbit_zoom_camera.rs]: I should probably make an effort to not assume 'f32' everywhere in here?

File: src/camera.rs (hunk 1 above)
Comment (bvssvni) [inline src/camera.rs]: cgmath might be deprecated in favor of nalgebra, and Piston uses vecmath which is designed for use in libraries (type agnostic, easy to reason about). Perhaps we should make a simple quaternion library?

File: src/orbit_zoom_camera.rs (hunk 2 above)
Comment (bvssvni) [inline src/orbit_zoom_camera.rs]: Being able to change between `f32` and `f64` is nice, but this can easily be fixed later.

**Comment by bvssvni** [comment]
I'm going to merge this because the issues addresses takes time to solve, and gets in the way of adding useful features. Will open up new issues to address these concerns.

**Comment by bvssvni** [comment]
Sorry! It seems that cgmath doesn't work with Rust nightly and the commit had to be reverted.

I've opened up a discussion about making a simple quaternion library https://github.com/PistonDevelopers/piston/issues/840

**Pull request merged**